geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/CreateAlterDestroyRegionCommands.java

```diff
@@ -907,6 +903,31 @@ private void validateRegionFunctionArgs(Cache cache, RegionFunctionArgs regionFu
                 new Object[] {regionFunctionArgs.getCompressor()}));
       }
     }
+
+    if (regionFunctionArgs.hasPartitionAttributes()) {
+      boolean partitionResolverFailure = false;
+      if (regionFunctionArgs.isPartitionResolverSet()) {
+        String partitionResolverClassName = regionFunctionArgs.getPartitionResolver();
+        Object partitionResolver = null;
+
+        try {
+          Class<?> compressorClass =
+              (Class<?>) ClassPathLoader.getLatest().forName(partitionResolverClassName);
+          partitionResolver = compressorClass.newInstance();
+        } catch (InstantiationException e) {
+          partitionResolverFailure = true;
+        } catch (IllegalAccessException e) {
+          partitionResolverFailure = true;
+        } catch (ClassNotFoundException e) {
+          partitionResolverFailure = true;
+        }
+        if (partitionResolverFailure || !(partitionResolver instanceof PartitionResolver)) {
+          throw new IllegalArgumentException(
```

Comment:
Use cause from above (you don't need to check it for null because null value is same as not having a cause):

 +          throw new IllegalArgumentException(
 +              CliStrings.format(CliStrings.CREATE_REGION__MSG__INVALID_PARTITION_RESOLVER,
 +                  new Object[] {regionFunctionArgs.getCompressor()}), cause);
 +        }